readthedocs_ext/readthedocs.py

```diff
@@ -261,99 +292,18 @@ def dump_sphinx_data(app, exception):
-    REPLACEMENT_PATTERN = re.compile(
-        r'''
-        ^\$\(document\).ready\(function\s*\(\)\s*{(?:\n|\r\n?)
-        \s*Search.init\(\);(?:\n|\r\n?)
-        \}\);
-        ''',
-        (re.MULTILINE | re.VERBOSE)
-    )
-
-    def get_static_readthedocs_context(self):
-        ctx = super(HtmlBuilderMixin, self).get_static_readthedocs_context()
-        if self.indexer is not None:
-            ctx.update(self.indexer.context_for_searchtool())
-        return ctx
-
-    def copy_static_readthedocs_files(self):
-        super(HtmlBuilderMixin, self).copy_static_readthedocs_files()
-        self._copy_searchtools()
-
-    def _copy_searchtools(self, renderer=None):
-        """Copy and patch searchtools
-
-        This uses the included Sphinx version's searchtools, but patches it to
-        remove automatic initialization. This is a fork of
-        ``sphinx.util.fileutil.copy_asset``
-        """
-        log.info(bold('copying searchtools... '), nonl=True)
-
-        if sphinx.version_info < (1, 8):
-            search_js_file = 'searchtools.js_t'
-        else:
-            search_js_file = 'searchtools.js'
-        path_src = os.path.join(
-            package_dir, 'themes', 'basic', 'static', search_js_file
-        )
-        if os.path.exists(path_src):
-            path_dest = os.path.join(self.outdir, '_static', 'searchtools.js')
-            if renderer is None:
-                # Sphinx 1.4 used the renderer from the existing builder, but
-                # the pattern for Sphinx 1.5 is to pass in a renderer separate
-                # from the builder. This supports both patterns for future
-                # compatibility
-                if sphinx.version_info < (1, 5):
-                    renderer = self.templates
-                else:
-                    from sphinx.util.template import SphinxRenderer
-                    renderer = SphinxRenderer()
-            with codecs.open(path_src, 'r', encoding='utf-8') as h_src:
-                with codecs.open(path_dest, 'w', encoding='utf-8') as h_dest:
-                    data = h_src.read()
-                    data = self.REPLACEMENT_PATTERN.sub(self.REPLACEMENT_TEXT, data)
-                    h_dest.write(renderer.render_string(
-                        data,
-                        self.get_static_readthedocs_context()
-                    ))
-        else:
-            log.warning('Missing {}'.format(search_js_file))
-        log.info('done')
-
-
-class ReadtheDocsBuilder(HtmlBuilderMixin, StandaloneHTMLBuilder):
-    name = 'readthedocs'
-
-
-class ReadtheDocsDirectoryHTMLBuilder(HtmlBuilderMixin, DirectoryHTMLBuilder):
-    name = 'readthedocsdirhtml'
-
-
-class ReadtheDocsSingleFileHTMLBuilder(BuilderMixin, SingleFileHTMLBuilder):
-    name = 'readthedocssinglehtml'
+class ReadtheDocsSingleFileHTMLBuilderLocalMedia(SingleFileHTMLBuilder):
 
+    """Sphinx builder that builds a single HTML file that will be zipped by Read the Docs."""
```

Comment:
Is there a reason we're keeping this overriden builder?

Reply:
I don't have a good reason. Both `ReadtheDocsSingleFileHTMLBuilder` and `ReadtheDocsSingleFileHTMLBuilderLocalMedia` were identical except in name. RTD uses both for slightly different purposes but I'm not sure if there's a reason for that.

Reply:
I believe `ReadtheDocsSingleFileHTMLBuilder` is used when a user wants to build their main docs as singlehtml: https://github.com/readthedocs/readthedocs.org/blob/8e78d680d02aeba12644796b979ef62459f64932/readthedocs/doc_builder/backends/sphinx.py#L335

and the localmedia is used for _our_ downloadable singlehtml builder. It's definitely confusing :)

Reply:
I see that. However, they're exactly the same. We could just use the built-in builder and zip it up.

Reply:
Yea, I think that should work fine. 👍

Reply:
Actually I think we do want a custom builder for downloadable docs. For downloadable docs, we don't want to add any additional JS (eg. `update_body` should do nothing). Using a separate builder for that seems fine since that is a special RTD-only feature.